Library/Homebrew/test/support/helper/spec/shared_context/integration_test.rb

```diff
@@ -174,15 +175,25 @@ def install
       RUBY
     end
 
-    Formulary.find_formula_in_tap(name.downcase, tap).tap do |formula_path|
-      formula_path.write <<~RUBY
+    formula_path = Formulary.find_formula_in_tap(name.downcase, tap).tap do |path|
+      path.write <<~RUBY
         class #{Formulary.class_s(name)} < Formula
         #{content.gsub(/^(?!$)/, "  ")}
         end
       RUBY
 
       tap.clear_cache
     end
+
+    unless tab.nil?
+      formula = Formula[name]
+      keg = formula.prefix
+      keg.mkpath
+      tab = Tab.new({ "tabfile" => keg/Tab::FILENAME }.merge(tab))
+      tab.write
+    end
```

Comment:
#Suggestion: You can probably simplify this by using `Tab.for_name`. Here's a tangentially related example.

https://github.com/Homebrew/brew/blob/8aef364d1e9bb074799897d599aee5c31e4d0fa7/Library/Homebrew/test/cmd/autoremove_spec.rb#L17-L21

I also seem to remember that this pattern of fake installing a formula by writing its tab was controversial but as you can see it's already a pattern used in other parts of the test suite.

Reply:
> I also seem to remember that this pattern of fake installing a formula by writing its tab was controversial but as you can see it's already a pattern used in other parts of the test suite.

Perhaps using `install_test_formula` here would make more sense than `setup_test_formula`?

https://github.com/Homebrew/brew/blob/f8b1d1f56cb19212b89e385e000ca3921912abeb/Library/Homebrew/test/support/helper/spec/shared_context/integration_test.rb#L188-L195

Reply:
IIRC we've faked the install in other places for performance reasons. If performance is not a concern or the difference is negligible, installing the test formula is preferable.

Reply:
It looks to be quite slow with the integration tests as @MikeMcQuaid mentioned below so reverting to the original approach is probably for the best.

```console
$ brew tests --only=cmd/mark --profile=5
Randomized with seed 20704
/usr/local/Homebrew/Library/Homebrew/api.rb:9: warning: base64 was loaded from the standard library, but will no longer be part of the default gems since Ruby 3.4.0. Add base64 to your Gemfile or gemspec.
Run options: exclude {:needs_linux=>true, :needs_network=>true, :needs_ci=>true, :needs_svn=>true}

Randomized with seed 20704
....

Top 4 slowest examples (19.45 seconds, 100.0% of total time):
  Homebrew::Cmd::Mark integration tests marks or unmarks a formula as installed on request
    7.59 seconds ./test/cmd/mark_spec.rb:34
  Homebrew::Cmd::Mark integration tests marks or unmarks a formula as installed as dependency
    6.96 seconds ./test/cmd/mark_spec.rb:56
  Homebrew::Cmd::Mark integration tests raises an error when a formula is not installed
    4.83 seconds ./test/cmd/mark_spec.rb:78
  Homebrew::Cmd::Mark behaves like parseable arguments can parse arguments
    0.0682 seconds ./test/cmd/shared_examples/args_parse.rb:8

Finished in 19.45 seconds (files took 4.08 seconds to load)
4 examples, 0 failures

Randomized with seed 20704
```

To be fair though these tests look slower since I've just run them on my old iMac but you get the idea.

Reply:
Gotcha, thanks!